platform/util/platform-util/src/test/java/org/codice/ddf/platform/util/properties/PropertiesLoaderTest.java

```diff
@@ -0,0 +1,200 @@
+
+  @Mock private Properties propertiesMock;
+
+  @ClassRule public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private static final String PROPERTIES_FILENAME = "test.properties";
+  private static final String NON_EXISTENT_FILENAME = "nonExistentFile.properties";
+  private static File propertiesFile;
+
+  private static Map<Object, Object> exampleMap = ImmutableMap.of("a", "A", "b", "B", "c", "C");
+
+  private static final PropertiesLoader PROPERTIES_LOADER = PropertiesLoader.getInstance();
+
+  @BeforeClass
+  public static void setUpClass() throws Exception {
+
+    // set up string to put in properties file
+    StringBuilder sb = new StringBuilder();
+    for (Map.Entry<Object, Object> entry : exampleMap.entrySet()) {
+      sb.append(entry.getKey() + "=" + entry.getValue() + "\n");
+    }
+
+    // initialize file
+    propertiesFile = temporaryFolder.newFile(PROPERTIES_FILENAME);
+    Files.write(propertiesFile.toPath(), sb.toString().getBytes());
+  }
+
+  @Test
+  public void testToMapWithProperties() throws Exception {
+    when(propertiesMock.entrySet()).thenReturn(exampleMap.entrySet());
+
+    Map<?, ?> entries = PROPERTIES_LOADER.toMap(propertiesMock);
+    Assert.assertEquals("Returns a map of the properties", exampleMap, entries);
+  }
+
+  @Test
+  public void testToMapWithNull() throws Exception {
+    Map<?, ?> entries = PROPERTIES_LOADER.toMap(null);
+    Assert.assertEquals("Returns an empty map", Collections.emptyMap(), entries);
+  }
+
+  @Test
+  public void testToMapWithEmptyProperties() throws Exception {
+    Map<?, ?> entries = PROPERTIES_LOADER.toMap(new Properties());
+    Assert.assertEquals("Returns an empty map", Collections.emptyMap(), entries);
+  }
+
+  @Test
+  public void testLoadPropertiesWithoutPropertiesFile() throws Exception {
+    Properties testProperties = PROPERTIES_LOADER.loadProperties(null, null);
+    Properties emptyProperties = new Properties();
+
+    Assert.assertEquals("Returns empty properties", emptyProperties, testProperties);
+  }
+
+  @Test
+  public void testLoadPropertiesWithEmptyPropertiesFile() throws Exception {
+    File emptyFile = temporaryFolder.newFile("tempFile");
+    Properties testProperties = PROPERTIES_LOADER.loadProperties(emptyFile.getPath(), null);
+    Properties emptyProperties = new Properties();
+
+    Assert.assertEquals("Returns empty properties", emptyProperties, testProperties);
+  }
+
+  @Test
+  public void testLoadPropertiesWithProperties() throws Exception {
+    Properties testProperties = PROPERTIES_LOADER.loadProperties(propertiesFile.getCanonicalPath());
+
+    Assert.assertEquals(
+        "Returns the correct properties", exampleMap.entrySet(), testProperties.entrySet());
+  }
+
+  @Test
+  public void testAttemptLoadWithSpring() throws Exception {
+    Properties testProperties =
+        PropertiesLoader.attemptLoadWithSpring(
+            PROPERTIES_FILENAME, this.getClass().getClassLoader());
```

Comment:
✏️  You could cache the class loader in memory:
```
private static final ClassLoader CLASS_LOADER = PropertiesLoader.class.getClassLoader();
```

Reply:
Eh. I only use it twice.